Autopilot/TelemetryManager/Src/telemetryStateClasses.cpp

```diff
@@ -95,18 +116,43 @@ telemetryState& readFromPathMode::getInstance()
 
 void analyzeDataMode::execute(telemetryManager* telemetryMgr)
 {
-    //set the dataValid here
+    
+    telemetryMgr->dataValid = 1;
+    telemetryMgr->dataValid &= sizeof(&_PMData) == 19;
+    telemetryMgr->dataValid &= &_PMData.beginLanding != nullptr;
```

Comment:
many of these fields are statically defined. Checking for `nullptr` will not work since they have a memory address. 

Instead you should check if their values fall within an acceptable range/set

Reply:
Pending ranges or sets